app/controllers/api/functions.php

```diff
@@ -698,15 +769,17 @@
         
         $deployment = $dbForProject->getDocument('deployments', $deploymentId);
         if ($deployment->isEmpty()) {
-            throw new Exception('Deployment not found', 404);
+            throw new Exception('Deployment not found', 404, Exception::DEPLOYMENT_NOT_FOUND);
         }
 
         if ($deployment->getAttribute('resourceId') !== $function->getId()) {
-            throw new Exception('Deployment not found', 404);
+            throw new Exception('Deployment not found', 404, Exception::DEPLOYMENT_NOT_FOUND);
         }
 
-        if (!$dbForProject->deleteDocument('deployments', $deployment->getId())) {
-            throw new Exception('Failed to remove deployment from DB', 500);
+        if ($deviceFunctions->delete($deployment->getAttribute('path', ''))) {
```

Comment:
We shouldn't be doing any delete operations here. The deletes worker should be doing it in the background

Reply:
is this handled in the background?